### DISCUSSION TITLE
Existing 3.0.0 App Can't Rebuild (No "exports" main defined...")

### DISCUSSION THREAD

**Discussion by zknicker** [nuxt/nuxt#19727] [General]
I have an existing Nuxt App hosted on Vercel that cannot redeploy today (aka I've made no changes to the code) due to the following build error:

```
[20:16:36.124] > build
[20:16:36.124] > nuxi build
[20:16:36.196] [log] Nuxi 3.0.0
[20:16:36.205] [log] Nuxt 3.0.0 with Nitro 1.0.0
[20:16:36.205] [error] No "exports" main defined in /vercel/path0/node_modules/@nuxt/kit/package.json
[20:16:36.206]   at new NodeError (internal/errors.js:322:7)
[20:16:36.206]   at throwExportsNotFound (internal/modules/esm/resolve.js:332:9)
[20:16:36.206]   at packageExportsResolve (internal/modules/esm/resolve.js:523:7)
[20:16:36.206]   at resolveExports (internal/modules/cjs/loader.js:450:36)
[20:16:36.206]   at Function.Module._findPath (internal/modules/cjs/loader.js:490:31)
[20:16:36.206]   at Function.Module._resolveFilename (internal/modules/cjs/loader.js:888:27)
[20:16:36.206]   at Function.resolve (internal/modules/cjs/helpers.js:107:19)
[20:16:36.206]   at resolveModule (node_modules/nuxi/dist/shared/nuxi.e551a86b.mjs:15:29)
[20:16:36.206]   at importModule (node_modules/nuxi/dist/shared/nuxi.e551a86b.mjs:35:24)
[20:16:36.206]   at loadKit (node_modules/nuxi/dist/shared/nuxi.8cc4a579.mjs:5:18)
[20:16:36.224] Error: Command "npm run build" exited with 1
[20:16:36.907] BUILD_FAILED: Command "npm run build" exited with 1
```

I've hunted around Discord and GitHib Issues, but haven't found any canonical tracking link for the issue. 

The reports are also far-ranging:
* Discord commenter suggests the issue is transitory: https://discord.com/channels/473401852243869706/1085613367458279514/1085613367458279514
* 2.X Nuxt apps suddenly being unable to deploy due to the same issue: https://github.com/nuxt/nuxt/issues/19717

For the record: I cannot build locally due to the same issue, despite upgrading to various flavors of Nuxt 3.X.

Anyone have any suggestions on a path to resolution, similar experiences, etc?

**Comment by danielroe** [comment]
Would you provide a reproduction? I would suggest you ensure you are updated to the latest version of Nuxt 3, and refresh your lockfile. (I suspect what is happening is that you are using Nuxt v3.0.0 but have a newer version of Nuxt Kit, such as v3.3, which is hoisted higher up in your dependency tree.)

**Reply by zknicker** [reply]
@danielroe thanks for the response. Here's a minimal reproduction: https://stackblitz.com/edit/nuxt-starter-zkfbgb?file=package.json. The only changes I made to StackBlitz's Nuxt3 template was the addition of the pinia and @pinia/nuxt dependencies.

**Reply by zknicker** [reply]
@danielroe so a couple notes here:

1. I'm able to fix the above issue by just upgrading Nuxt, Pinia, and @Pinia/Nuxt to their latest versions.
2. Doing so breaks several other things in my project. Most notably, I began getting this new error `Failed to resolve import "fsevents" from "node_modules/.vite/deps/chunk-B47FSDGB.js?v=73bb0042`. I tried to get a working StackBlitz repro for this and got down to a combination of Nuxt, Pinia, Tailwind, and a Vite plugin. Ultimately the fix was removing `./nuxt.config.{js,ts}` from Tailwind's `Content` config. Super bizarre. You can repro this and the fix here: https://stackblitz.com/edit/nuxt-starter-zkfbgb?file=tailwind.config.js.
3. Luxon also somehow broke amidst all this. I was previously using `Luxon 2.x` and importing it as a .js. I just upgraded to `Luxon 3.x` and now import it as an es6 module, and everything seems to be working there.

So long story short, you can probably save your time on all of the above. I'm just surprised that a new Nuxt release broke so much in an otherwise working production app. Do you have any advice for avoiding these problems in the future? Is this an expected set of problems for someone who pins their Nuxt version to 3.0.0 :)?

**Reply by danielroe** [reply]
Likely, the issue was that Vercel was adding `@nuxtjs/web-vitals` to your project, which was overriding your lockfile and installing a newer version of `@nuxt/kit` in your project: https://github.com/nuxt/nuxt/issues/19717#issuecomment-1472331600. We've hotfixed it so, had you not upgraded everything (but well done!), hopefully your issue would have been solved.
